lib/liquid/static_registers.rb

```diff
@@ -1,36 +1,35 @@
 # frozen_string_literal: true
 
+require 'forwardable'
+
 module Liquid
   class StaticRegisters
+    extend Forwardable
+
     attr_reader :static, :registers
 
+    def_delegators :@cache, :[], :key?
```

Comment:
Here in https://github.com/Shopify/liquid/pull/1250, the PR become to simply add the method here and to :fire: custom logic.

Reply:
If you are trying to avoid `*args` looks like `def_delegators` has the same problem (https://github.com/ruby/ruby/blob/38a4f617de157586668dd726d518eadcebf1bca2/lib/forwardable.rb#L207-L233), although perhaps that could be fixed upstream to leverage `...`

Reply:
I was actually trying to avoid duplicating logic and making sure StaticRegister methods that were mimicking Hash behaved the same way.

I'm open to do what is needed to make iterative changes (I could stick to the first commit of this PR)